cowbird/permissions_synchronizer.py

```diff
@@ -74,59 +74,178 @@ def __init__(self,
+                    # match any name with specific type, 0 or more times
+                    res_regex += rf"(/\w+:{segment['type']})*"
+                else:
+                    # match name and type exactly
+                    res_regex += f"/{segment['name']:}:{segment['type']}"
+                    match_len += 1
+            res_regex += "$"
+            if re.match(res_regex, resource_nametype_path):
+                matched_res_dict[res_key] = match_len
+
+        # Find the longest match
+        max_matching_keys = [res for res, match_len in matched_res_dict.items()
+                             if match_len == max(matched_res_dict.values())]
+        if len(max_matching_keys) == 1:
+            src_res_key = max_matching_keys[0]
+        elif len(max_matching_keys) > 1:
+            raise ValueError("Found 2 matching resources of the same length in the config. Ambiguous config resources :"
+                             f" {max_matching_keys}")
+        else:
+            raise ValueError("No matching resources could be found in the config file.")
+        return src_res_key, matched_res_dict[src_res_key]
+
+    @staticmethod
+    def _create_res_data(target_segments, src_resource_suffix):
+        # type: (List[Dict], List[Dict]) -> List[Dict]
+        """
+        Creates resource data used to update permissions. This data includes the name and type of each segments of
+        a full resource path.
+
+        @param target_segments List containing the name and type info of each segment of the target resource path.
+        @param src_resource_suffix List similar to the `target_res` argument, but for the input source resource. This
+                                   list contains only the suffix section of the resource path, which is the part that is
+                                   common to both source and target resource paths.
+        """
+        permissions_data = []
+        suffix_target_segments = []
+        # First add 'named' resource data
+        for i, segment in enumerate(target_segments):
+            if segment["name"] in [SINGLE_TOKEN, MULTI_TOKEN]:  # pylint: disable=no-else-break
+                suffix_target_segments = target_segments[i:]
+                break
+            else:
+                permissions_data.append({
+                    "resource_name": segment["name"],
+                    "resource_type": segment["type"]
+                })
+        # Then add 'tokenenized' resource data, if any
+        if suffix_target_segments:
+            # Make regex for the tokenized part of the target resource
+            suffix_regex = "^"
+            for segment in suffix_target_segments:
+                if segment["name"] == SINGLE_TOKEN:
+                    # match 1 name only
+                    suffix_regex += r"(/\w+)"
+                elif segment["name"] == MULTI_TOKEN:
+                    # match any name 0 or more times
+                    suffix_regex += r"((?:/\w+)*)"
```

Comment:
same potential issue for `\w+` here
maybe consider using a `rf` string with a constant that defines the resource pattern definition